server/plugins/task-plugin-api/src/main/java/com/oceanbase/odc/plugin/task/api/partitionplan/AutoPartitionExtensionPoint.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.plugin.task.api.partitionplan;
+
+import java.sql.Connection;
+import java.util.List;
+
+import org.pf4j.ExtensionPoint;
+
+import com.oceanbase.odc.plugin.task.api.partitionplan.invoker.create.PartitionExprGenerator;
+import com.oceanbase.odc.plugin.task.api.partitionplan.invoker.drop.DropPartitionGenerator;
+import com.oceanbase.odc.plugin.task.api.partitionplan.invoker.partitionname.PartitionNameGenerator;
+import com.oceanbase.tools.dbbrowser.model.DBTable;
+import com.oceanbase.tools.dbbrowser.model.DBTablePartition;
+import com.oceanbase.tools.dbbrowser.model.datatype.DataType;
+
+import lombok.NonNull;
+
+/**
+ * {@link AutoPartitionExtensionPoint}
+ *
+ * @author yh263208
+ * @date 2023-01-11 20:23
+ * @since ODC_release_4.2.4
+ * @see org.pf4j.ExtensionPoint
+ */
+public interface AutoPartitionExtensionPoint extends ExtensionPoint {
+
+    boolean supports(@NonNull DBTablePartition dbTablePartition);
+
+    String unquoteIdentifier(@NonNull String identifier);
```

Comment:
two differences:
1. StringUtil is just a tool, we should not use it at the service layer, cause that we don't know which dialect type is so that we don't know which method should we call (you may think that we can get dialect type and use if-else block to do that, but this is too customized).
2. we should process upper-lower case problem, that is to say, if oracle identifiers are like these `"aaa"`, `aaa`,  and we can get new identifiers by calling this method: `aaa`, `AAA`.